### PR TITLE
 hooks: tkinter: avoid collecting data from system Tcl/Tk framework on macOS

### PR DESCRIPTION
**Pull request by rokm**
After a discussion with @bwoodsend, we reached conclusion that in the case of system Tcl/Tk framework on macOS, it makes no sense to collect data when we do not collect the shared library (which we do not; and is not needed anymore as #5172 ensures that the system libraries are used correctly). So this PR removes the collection of data from macOS' system Tcl/Tk. This affects only python builds that use system Tcl/Tk - older python.org builds (e.g., 32/64-bit 3.6.5) and apparently homebrew python.

As a bonus, this also takes care of the error in `tkinter` hook with homebrew python on macOS 11, which was reported in #5040.

-----

On macOS, we do not collect system libraries in general; if `tkinter` uses system Tcl/Tk 8.5 framework (older 32/64-bit
python.org builds, or brew python), we do not collect its Tk and Tcl shared library, either. Therefore, it makes little sense to
collect the data (scripts and modules) from the system framework, either.

This commit modifies the behavior of both `_tkinter` stdhook and rthook on macOS. The stdhook now treats `(None, None)` returned by `_find_tcl_tk()` as an indicator that the system framework is used and that data directories should not be collected. The rthook now sets `TCL_LIBRARY` and `TK_LIBRARY` environment variables if collected tcl and tk directories exist, but does not raise error anymore if they do not (under assumption that they have not been collected from the system framework).

The behavior under other OSes remains unchanged.

### CODE REVIEW

File: PyInstaller/hooks/hook-_tkinter.py
Comment (bwoodsend) [inline PyInstaller/hooks/hook-_tkinter.py:271]: I suppose this is now a false positive on macOS11?

Comment (rokm) [reply PyInstaller/hooks/hook-_tkinter.py:271]: It will be printed on macOS 11 with homebrew python, yes. But it will actually be correct on both accounts - the system framework is used, and therefore the shared libraries are currently not found.

It will also be printed on earlier macOS versions with older python.org builds (e.g., 32/64-bit 3.6.5) or homebrew python.

**Review by bwoodsend — APPROVED** [review]
I feel like my review isn't worth much here seeing as I can't test it.

**Comment by Legorooj** [comment]
I'm going to wait until #5175 is merged to review this. Is that PR ready to merge @rokm or is there still work to be done?

**Comment by rokm** [comment]
> I'm going to wait until #5175 is merged to review this. Is that PR ready to merge @rokm or is there still work to be done?

It should be ready to merge - as far as I'm concerned, we should be collecting all those Tcl modules to ensure that packaged environment is the same as the source one.

**Review by Legorooj — APPROVED** [review]
Thanks for this @rokm

**Pull request merged**
